cmd/kubeadm/app/phases/etcd/local.go

```diff
@@ -223,6 +223,7 @@ func getEtcdCommand(cfg *kubeadmapi.ClusterConfiguration, endpoint *kubeadmapi.A
 		"peer-trusted-ca-file":        filepath.Join(cfg.CertificatesDir, kubeadmconstants.EtcdCACertName),
 		"peer-client-cert-auth":       "true",
 		"snapshot-count":              "10000",
+		"logger":                      "zap",
```

Comment:
https://github.com/etcd-io/etcd/issues/11426
if the default logger changes in a future release, why do we have to preemtively switch to the new logger in our deployments of older etcd?

/hold

Reply:
@jingyih does etcd 3.4 plan to set the default logger to zap? now, k8s already uses version 3.4,but logger is old, so trace feature cannot work,it plays a key role in trouble-shooting. if etcd 3.4 does not use zap, i think it is better if we explicitly set etcd --logger to zap.

Reply:
etcd 3.4 supports both the capnslog (old) and zap (new) logger. I think it is user's choice to pick either (we recommend zap). If the logger is not explicitly set to zap here, is there a way for user to provide customized / additional flags to etcd?

Reply:
IMO we should keep kubeadm's etcd.yaml static pod manifest without being explicit about the logger. currently kubeadm does support patching of the manifests using --experimental-kustomize, so the logger can be changed if the user really wants that.

Reply:
thanks.i see.

Reply:
zap is better than deprecated capnslog, it looks risk-free,why do not we set an optimal logger by default?  This is my pr motivation. thanks.

Reply:
i agree that this is mostly risk free, but kubeadm should not be opinionated about the loggers in etcd. if the users want to change it they can modify their etcd.yaml.